pkg/apis/networking/validation/validation.go

```diff
@@ -81,6 +81,48 @@ func ValidateNetworkPolicySpec(spec *networking.NetworkPolicySpec, fldPath *fiel
 			}
 		}
 	}
+	// Validate egress rules
```

Comment:
Other things we probably want to validate here:
- PolicyTypes is not empty
- The correct PolicyTypes are present for the given Ingress / Egress rules (i.e. can't have egress rules with no "Egress" item in the PolicyTypes list.